ui-frontend/packages/catalog-ui-search/src/main/webapp/js/model/Query.js

```diff
@@ -124,7 +124,7 @@ Query.Model = Backbone.AssociatedModel.extend({
     ) {
       // for backwards compatability
       try {
-        data.filterTree = JSON.parse(data.filterTree)
+        data.filterTree = new FilterBuilderClass(JSON.parse(data.filterTree))
```

Comment:
Looks like this is a wrapper for anytime we call set - so it ended up with this weird feedback loop where it would overwrite the filterTree created this way in the constructor.